kernel/src/table_properties/deserialize.rs

```diff
@@ -76,19 +76,36 @@ fn try_parse(props: &mut TableProperties, k: &str, v: &str) -> Option<()> {
         }
         "delta.checkpointPolicy" => props.checkpoint_policy = CheckpointPolicy::try_from(v).ok(),
         "delta.enableRowTracking" => props.enable_row_tracking = Some(parse_bool(v)?),
+        "delta.enableInCommitTimestamps" => {
+            props.enable_in_commit_timestamps = Some(parse_bool(v)?)
+        }
+        "delta.inCommitTimestampEnablementVersion" => {
+            props.in_commit_timestamp_enablement_version = Some(parse_int(v)?)
+        }
+        "delta.inCommitTimestampEnablementTimestamp" => {
+            props.in_commit_timestamp_enablement_timestamp = Some(parse_int(v)?.try_into().ok()?)
+        }
         _ => return None,
     }
     Some(())
 }
 
-/// Deserialize a string representing a positive integer into an `Option<u64>`. Returns `Some` if
-/// successfully parses, and `None` otherwise.
+/// Deserialize a string representing a positive (non-zero) integer into an `Option<u64>`. Returns
+/// `Some` if successfully parses, and `None` otherwise.
 pub(crate) fn parse_positive_int(s: &str) -> Option<NonZero<u64>> {
     // parse to i64 (then check n > 0) since java doesn't even allow u64
     let n: i64 = s.parse().ok()?;
     NonZero::new(n.try_into().ok()?)
 }
 
+/// Deserialize a string representing a positive integer into an `Option<u64>`. Returns `Some` if
+/// successfully parses, and `None` otherwise.
+pub(crate) fn parse_int(s: &str) -> Option<u64> {
```

Comment:
make it self-documenting?
```suggestion
pub(crate) fn parse_uint(s: &str) -> Option<u64> {
```